cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,26 +38,54 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
 
-	if !cmdUtil.IsValidProfile(profileName) {
+	if cmdUtil.IsValidProfile(profileName) {
+		//need to check if the current active profile has a VM
+		if cmdUtil.DoesVMExist(config.AllInstancesConfig.ActiveProfile) {
+			err := unsetCurrentOcContext(config.AllInstancesConfig.ActiveProfile)
+			if err != nil {
+				fmt.Println(err.Error())
+			}
+		}
+
 		err := profileActions.SetActiveProfile(profileName)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
-	} else {
-		err := cmdUtil.SetOcContext(profileName)
+		fmt.Printf("Profile '%s' set as active profile\n", profileName)
+
+		err = cmdUtil.SetOcContext(profileName)
 		if err != nil {
 			if glog.V(2) {
 				fmt.Println(fmt.Sprintf("%s", err.Error()))
 			}
-			fmt.Println(fmt.Sprintf("oc cli context could not changed for '%s'. %s", profileName, unableToSetOcContextErrorMessage))
+			fmt.Println(fmt.Sprintf("No oc CLI context is activated for profile, because the '%s' VM does not exist or is not running. Run 'minishift start' first.", profileName))
 		}
-
-		err = profileActions.SetActiveProfile(profileName)
+	} else {
+		err := profileActions.SetActiveProfile(profileName)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
+		fmt.Printf("Profile '%s' set as active profile\n", profileName)
+
+		if cmdUtil.DoesVMExist(config.AllInstancesConfig.ActiveProfile) {
+			err := unsetCurrentOcContext(config.AllInstancesConfig.ActiveProfile)
+			if err != nil {
+				fmt.Println(err.Error())
+			}
```

Comment:
This if block is not required and also it wont be called ever. As if it is not a valid profile the VM wont exist for sure. Profile list is generated from the actual directory structure, so If the VM exists then the profile directory will be present.

Reply:
actually the order is wrong, we need to remove the context of the previously active profile, it has to be done before we set the new profile as active profile.